modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -775,6 +781,35 @@ cv::gimpl::GFluidExecutable::GFluidExecutable(const ade::Graph &g,
         }
     }
 
+    return r;
+}
+
+
+cv::gimpl::GFluidExecutable::GFluidExecutable(const ade::Graph  &g,
+        const cv::gimpl::FluidGraphInputData                        &traverse_res,
+        const std::vector<cv::gapi::own::Rect>                  &outputRois)
+    : m_g(g), m_gm(m_g)
+{
+    GConstFluidModel fg(m_g);
+
+    std::size_t mat_count = 0;
+
+    std::tie(m_scratch_users, m_id_map, m_all_gmat_ids, mat_count) = std::tie(traverse_res.m_scratch_users, traverse_res.m_id_map, traverse_res.m_all_gmat_ids, traverse_res.m_mat_count);
+
+    for (auto const& agent_data : traverse_res.m_agents_data){
+        switch (agent_data.kind)
+        {
+        case GFluidKernel::Kind::Filter:    m_agents.emplace_back(new FluidFilterAgent(m_g, agent_data.nh));    break;
+        case GFluidKernel::Kind::Resize:    m_agents.emplace_back(new FluidResizeAgent(m_g, agent_data.nh));    break;
+        case GFluidKernel::Kind::NV12toRGB: m_agents.emplace_back(new FluidNV12toRGBAgent(m_g, agent_data.nh)); break;
+        default: GAPI_Assert(false);
+        }
+
+        auto& agent_ptr = m_agents.back();
+
+        std::tie(agent_ptr->in_buffer_ids, agent_ptr->out_buffer_ids) = std::tie(agent_data.in_buffer_ids, agent_data.out_buffer_ids);
+    }
+
     // Check that IDs form a continiuos set (important for further indexing)
     GAPI_Assert(m_id_map.size() >  0);
     GAPI_Assert(m_id_map.size() == static_cast<size_t>(mat_count));
```

Comment:
These asserts can be moved to `fluidExtractInputDataFromGraph`. And `mat_count` here seems to be redundant, `traverse_res.m_mat_count` can be tied to `m_num_int_buffers` directly

Reply:
fixed. Thanks